File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int n=nums.length;
        int i=0;
        int j=0;
        int idx=0;
        while(j<n){
            while(j<n && nums[i]==nums[j]) j++;
            nums[idx]=nums[i];
            idx++;
            i=j;
        }
        return idx;
    }
}

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n=nums.size();
        if(n==1) return 1;

        int i=0;
        int j=0;
        int k=0;
        while(j<n){
            while(j<n && nums[i]==nums[j]) j++;
            nums[k]=nums[i];
            i=j;
            k++;
        }
        
        return k;
    }
};